op-program/client/interop/cnosolidate.go

```diff
@@ -0,0 +1,227 @@
+package interop
+
+import (
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-program/client/interop/types"
+	"github.com/ethereum-optimism/optimism/op-program/client/l2"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/cross"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/depset"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/processors"
+	supervisortypes "github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+func ReceiptsToExecutingMessages(receipts ethtypes.Receipts) ([]*supervisortypes.ExecutingMessage, error) {
+	var execMsgs []*supervisortypes.ExecutingMessage
+	for _, rcpt := range receipts {
+		for _, l := range rcpt.Logs {
+			execMsg, err := processors.DecodeExecutingMessageLog(l)
+			if err != nil {
+				return nil, err
+			}
+			execMsgs = append(execMsgs, execMsg)
+		}
+	}
+	return execMsgs, nil
+}
+
+func Consolidate(deps ConsolidateCheckDeps,
+	oracle l2.Oracle,
+	transitionState *types.TransitionState,
+	superRoot *eth.SuperV1,
+) (eth.Bytes32, error) {
+	var consolidatedChains []eth.ChainIDAndOutput
+
+	for i, chain := range superRoot.Chains {
+		progress := transitionState.PendingProgress[i]
+		receipts := oracle.ReceiptsByBlockHash(progress.BlockHash, chain.ChainID)
```

Comment:
Do we need to set this up to send the hint that it may need to reproduce this block since it might not be canonical? Won't matter for the happy path but if pending is different to canonical the host will have to regenerate the optimistic block to populate the receipts.

Reply:
added a TODO to remember to do this when we can build deposits-only blocks.